chef/cookbooks/network/recipes/default.rb

```diff
@@ -90,7 +90,8 @@
   subscribes :run, resources(cookbook_file: "modprobe-bridge.conf"), :delayed
 end
 
-provisioner = search(:node, "roles:provisioner-server")[0]
+provisioner_instance = node[:provisioner][:config][:environment].gsub(/^provisioner-config-/, "") rescue "default"
+provisioner = node_search_with_cache("roles:provisioner-server", provisioner_instance).first
```

Comment:
I'm fine extracting the first line to a helper, but I don't think it makes sense to replace the two lines with a one-liner as this becomes too much voodoo if we do that, imho.

I'll add a patch tomorrow for the small helper.